index.js

```diff
@@ -25,21 +25,30 @@ document.addEventListener('DOMContentLoaded', () => {
   const textOut = document.getElementById('text-output');
   const fileIn = document.getElementById('file-input');
   const chooseFile = document.getElementById('file-button');
+  const splitIn = document.getElementById('period-split');
   const nameIn = document.getElementById('name-input');
   const teamIn = document.getElementById('team-input');
 
   const wordRegex = /\s*(\S{1,127})\s*/g;
 
   const rebuild = () => {
-    const pasta = text.value;
+    const pasta = splitIn.checked ? text.value.split('.') : [text.value];
     const lines = [];
-    let match = wordRegex.exec(pasta);
-    while (match != null) {
-      if (lines.length === 0 || lines[lines.length-1].length + match[1].length + 1 > 127)
-        lines.push(match[1]);
-      else lines[lines.length-1] += ' ' + match[1];
-      match = wordRegex.exec(pasta);
-    }
+
+    pasta.forEach(element => {
+      element = splitIn.checked && element.length ? element.trim() + '.' : element;
```

Comment:
I think the `element`s in `pasta` should have the periods appended just after splitting. That way `splitIn.checked` is only handled in one location. Separating the preprocessing also means this code can stay the same if new split methods are added.